tests/utils.go

```diff
@@ -2792,6 +2793,48 @@ func NewConsoleExpecter(virtCli kubecli.KubevirtClient, vmi *v1.VirtualMachineIn
 	}, timeout, opts...)
 }
 
+// The method adds the expect.BSnd command to the exect.BExp expression.
+// It is done to make sure the match was found in the result of the expect.BSnd
+// command and not in a leftover the wasn't removed from the buffer.
+func ExpectBatchWithValidatedSend(expecter expect.Expecter, batch []expect.Batcher, timeout time.Duration) ([]expect.BatchRes, error) {
```

Comment:
personally , i prefer that the fixes would be embedded in the current batches, 
i.e dont have double EXP, and wait for the right \n0  / prompt
or alternatively wrapping the command with the expected retcode (as discussed with Dan), and if there are special cases, handle them without generic method

as this approach might add some more nits that are need to be taken care of 
such as using \r instead of \n , maybe it was also before, but i didnt have this problem at least in #3543 

but its up to you, and the other reviewers
i am afraid that we will now have mixed approaches, and mixed use of \n or \r
batches that cant start with Snd \n (as you said but i am not sure we finished the discussion there)